### ISSUE TITLE
Add `if (foo == true)` to no-extra-boolean-cast or make it a separate rule

### ISSUE THREAD

**Issue by RDambrosio016** [rslint/rslint#68]
One pattern which ESLint does not seem to catch is
```js
if (foo == true) { /* */ }
```
no-extra-boolean-cast does not catch it, id like to make no-extra-boolean-cast encompass this pattern too, but that makes the name kind of misleading, so i am not sure whether we should rename it, and rename it to what exactly. Maybe we should make this a new rule?

**Comment by cwkang1998** [comment]
Can I try working on this? Maybe we can name it something like `no-extra-boolean-compare`?

**Comment by cwkang1998** [comment]
@Stupremee If we are not planning to proceed with this, should we close the issue?

**Comment by Stupremee** [comment]
Ah yes. Thank you for the reminder!
